### PR TITLE
OC 10732 Ability to run a report configured in Insight from within OpenClinica

### PR DESCRIPTION
**Pull request by theywa**
https://jira.openclinica.com/browse/OC-10732

### CODE REVIEW

File: web/src/main/java/org/akaza/openclinica/controller/InsightReportController.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            labels[0] = label;
+            labels[1] = label + replicaSubstring;
+        }
+        return labels;
+    }
+
+    public String doPost(String api, String method, HashMap<String, String> headers,
+                         String body) throws Exception {
+        URL url = new URL(api);
+        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
```
Comment (yogi-oc) [inline web/src/main/java/org/akaza/openclinica/controller/InsightReportController.java]: Can you please use RestTemplate? See examples in other places in the project.

Comment (theywa) [reply web/src/main/java/org/akaza/openclinica/controller/InsightReportController.java]: code updated

File: web/src/main/java/org/akaza/openclinica/controller/InsightReportController.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        } catch (IOException e) {
+            System.out.println("=== Error create file : " + e.getMessage());
+            e.printStackTrace();
+        } finally {
+            try {
+                if (fop != null) {
+                    fop.close();
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
```
Comment (yogi-oc) [inline web/src/main/java/org/akaza/openclinica/controller/InsightReportController.java]: Please log the exception. Do not print it.
Log it like this:
logger.error("Some message..." + e);

File: web/src/main/java/org/akaza/openclinica/controller/InsightReportController.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            RestTemplate rest = new RestTemplate();
+            ResponseEntity<String> response;
+            if (queries != null) {
+                response = rest.postForEntity(api, request, String.class, queries);
+            } else {
+                response = rest.postForEntity(api, request, String.class);
+            }
+            return response.getBody();
+        } catch (Exception e) {
+            System.out.println("## Error :: "+e.getMessage());
```
Comment (yogi-oc) [inline web/src/main/java/org/akaza/openclinica/controller/InsightReportController.java]: Please log the exception. Do not print it.
Log it like this:
logger.error("Some message..." + e);

File: web/src/main/java/org/akaza/openclinica/controller/InsightReportController.java (hunk 2 above)
Comment (theywa) [reply web/src/main/java/org/akaza/openclinica/controller/InsightReportController.java]: code updated

**Pull request merged**